File: src/app/entities/rides/ride.ts
import { DatePipe, Time } from '@angular/common';
import { FlightCompany } from '../flights/flight-company';
import { RideCompany } from './ride-company';
import { Location } from '../flights/location';

export class Ride {
    id: number
    company: RideCompany;
    companyID: number
    carMaker: string;
    carModel: string;
    productionYear: string;
    numberOfSeats: number;
    carType: string;
    locationID:number
    location:Location

    constructor() {

    }
}

File: projectBackend/projectBackend/Models/Ride.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace projectBackend.Models
{
  public class Ride
  {   
    [Key]
    public int ID { get; set; }
    public string CarMaker { get; set; }
    public string CarModel { get; set; }
    public string CarType { get; set; }
    public string ProductionYear { get; set; }
    public int NumberOfSeats { get; set; }
    public double price { get; set; }
    public int CompanyID { get; set; }
    [ForeignKey("CompanyID")]
    public virtual RideCompany Company { get; set; }

    public int LocationID { get; set; }
    [ForeignKey("LocationID")]
    public virtual Location Location { get; set; }

    public virtual List<Reservation> Reservations { get; set; }
  }
}
